pos_accounting/src/pool/operations.rs

```diff
@@ -73,6 +78,7 @@ pub enum PoSAccountingUndo {
     CreateDelegationId(CreateDelegationIdUndo),
     DelegateStaking(DelegateStakingUndo),
     SpendFromShare(SpendFromShareUndo),
+    IncreasePoolBalance(IncreasePoolBalanceUndo),
```

Comment:
Just for completeness, keep in mind that in the future we will need a `DecreasePoolbalance` thing too.

I thought of combining them, but then I changed my mind. Perhaps keeping them separated makes testing easier.

Reply:
Do you mean explicit Decrease and not undo of Increase? Curious what for? Penalties for misbehavior?

Reply:
Cancelling delegation.

Reply:
we have `PoSAccountingOperations::spend_share_from_delegation_id` for that